rosbag2_test_common/include/rosbag2_test_common/tested_storage_ids.hpp

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022, Foxglove Technologies Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef ROSBAG2_TEST_COMMON__TESTED_STORAGE_IDS_HPP_
+#define ROSBAG2_TEST_COMMON__TESTED_STORAGE_IDS_HPP_
+#include <vector>
+#include <string>
+
+
+namespace rosbag2_test_common
+{
+static const std::vector<std::string> kTestedStorageIDs = {
```

Comment:
Just do this as a `map` from ID to extension, include the `get_extension` function in this header - only specify the list of plugins once that way, specify data rather than code

Reply:
This isn't simple, the `testing::ValuesIn` function called in INSTANTIATE_TEST_P doesn't iterate across maps easily. I think i'd prefer not to keep it a map.